sdk/core/System.ClientModel/src/Options/PipelineMessageClassifier.cs

```diff
@@ -5,17 +5,84 @@
 {
+    /// <summary>
+    /// A default classifier instance. This classifier will classify a
+    /// <see cref="PipelineResponse"/> with a status code of <c>4xx</c> or
+    /// <c>5xx</c> as an error, and with a status code of <c>408</c>,
+    /// <c>429</c>, <c>500</c>, <c>502</c>, <c>503</c> and <c>504</c> as
+    /// retriable.
+    /// </summary>
     public static PipelineMessageClassifier Default { get; } = new EndOfChainClassifier();
 
+    /// <summary>
+    /// Create an instance of a <see cref="PipelineMessageClassifier"/> from a
+    /// collection of success status codes.
+    /// </summary>
+    /// <param name="successStatusCodes">The status codes that the returned
+    /// classifier instance will classifiy as success codes.</param>
+    /// <returns>A <see cref="PipelineMessageClassifier"/> instance that
+    /// classifies the status codes provided by
+    /// <paramref name="successStatusCodes"/> as success codes.</returns>
     public static PipelineMessageClassifier Create(ReadOnlySpan<ushort> successStatusCodes)
         => new ResponseStatusClassifier(successStatusCodes);
 
-    protected PipelineMessageClassifier() { }
-
+    /// <summary>
+    /// Attempt to evaluate whether the provided <see cref="PipelineMessage"/>
+    /// contains a <see cref="PipelineMessage.Response"/> that the client should
+    /// consider an error response.
+    /// </summary>
+    /// <param name="message">The <see cref="PipelineMessage"/> to classify.</param>
+    /// <param name="isError"><c>true</c> if the
+    /// <see cref="PipelineMessage.Response"/> should be considered an error
+    /// response.</param>
+    /// <returns><c>true</c> if the classifier had an opinion regarding whether
+    /// the service response was an error response; <c>false</c> otherwise.
+    /// </returns>
+    /// <remarks>
+    /// Not all classifiers derived from <see cref="PipelineMessageClassifier"/>
+    /// will classify a given <see cref="PipelineMessage"/>. Returning
+    /// <c>false</c> from <see cref="TryClassify(PipelineMessage, out bool)"/>
+    /// allows a classifier instance to compose with other classifiers by
+    /// passing the classification decision to a later instance.
+    /// <see cref="Default"/> will always return <c>true</c> from
+    /// <see cref="TryClassify(PipelineMessage, out bool)"/> and may be used as
+    /// the last classifier in any composed classifier collection.
+    /// </remarks>
     public abstract bool TryClassify(PipelineMessage message, out bool isError);
 
+    /// <summary>
+    /// Attempt to evaluate whether the provided <see cref="PipelineMessage"/>
+    /// contains a <see cref="PipelineMessage.Response"/> that indicates the
+    /// client should retry the <see cref="PipelineMessage.Request"/>.
+    /// </summary>
+    /// <param name="message">The <see cref="PipelineMessage"/> to classify.</param>
+    /// <param name="exception">An <see cref="Exception"/>, if any, that will
```

Comment:
I'm not sure from the context of this doc what the role this exception plays in the classification.   We may want to consider adding some detail.